### PR TITLE
Naive Bayes with Differential Privacy

### PR DESCRIPTION
**Pull request by grilhami**
## Description
Proof of concept of machine learning with differential privacy. Current implementation involves a differentially private Naive Bayes with Laplace and Geometric mechanism. Further functionalities are added such as validation for epsilon and delta values and privacy budget.

## Affected Dependencies
Scikit-learn is one of the primary requirements for the implementation since it involves inheriting from scikit-learn's Naive Bayes class.

## How has this been tested?
The best way to see if the implementation works are by the following example, which is similar to Scikit-learn's example :
```
from sklearn import datasets
from sklearn.model_selection import train_test_split

from pydp.ml.naive_bayes import GaussianNB

dataset = datasets.load_iris()
X_train, X_test, y_train, y_test = train_test_split(dataset.data, dataset.target, test_size=0.2)

clf = GaussianNB()
clf.fit(X_train, y_train)
```

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: tests/algorithms/test_bounded_mean/test_bounded_mean_int64_data.proto [hunk 1]
```diff
@@ -0,0 +1,2 @@
+M
```
Comment (chinmayshah99) [inline tests/algorithms/test_bounded_mean/test_bounded_mean_int64_data.proto]: Can we remove these special characters?

File: src/pydp/ml/util/accountant.py [hunk 2]
```diff
@@ -0,0 +1,372 @@
+from numbers import Integral
```
Comment (chinmayshah99) [inline src/pydp/ml/util/accountant.py:22]: Can we please add the source file link along with its license?

**Review by chinmayshah99 — CHANGES_REQUESTED** [review]

**Comment by chinmayshah99** [comment]
Merging it into the branch as the draft is ready and we can do a further development on this.
